# # Desafío 2 - Estadistica descriptiva y probabilidades (Parte II)

En primer lugar importamos las librerias necesarias

In [144]:
import pandas as pd

Cargamos el dataframe worldcup2014.csv

In [145]:
df_worldcup = pd.read_csv('worldcup2014.csv',delimiter=',')
df_worldcup.head(32)

team     continent group  group_pos  cantidad_juegos  \
0                   Brazil  southamerica     A          1                3   
1                   Mexico  northamerica     A          2                3   
2                  Croatia        europe     A          3                3   
3                 Cameroon        africa     A          4                3   
4              Netherlands        europe     B          1                3   
5                    Chile  southamerica     B          2                3   
6                    Spain        europe     B          3                3   
7                Australia          asia     B          4                3   
8                 Colombia  southamerica     C          1                3   
9                   Greece        europe     C          2                3   
10             Ivory Coast        africa     C          3                3   
11                   Japan          asia     C          4                3   
12              Costa Rica  northamerica     D          1                3   
13                 Uruguay  southamerica     D          2                3   
14                   Italy        europe     D          3                3   
15                 England        europe     D          4                3   
16                  France        europe     E          1                3   
17             Switzerland        europe     E          2                3   
18                 Ecuador  southamerica     E          3                3   
19                Honduras  northamerica     E          4                3   
20               Argentina  southamerica     F          1                3   
21                 Nigeria        africa     F          2                3   
22  Bosnia and Herzegovina        europe     F          3                3   
23                    Iran          asia     F          4                3   
24                 Germany        europe     G          1                3   
25           United States  northamerica     G          2                3   
26                Portugal        europe     G          3                3   
27                   Ghana        africa     G          4                3   
28                 Belgium        europe     H          1                3   
29                 Algeria        africa     H          2                3   
30                  Russia        europe     H          3                3   
31             South Korea          asia     H          4                3   

    juegos_ganados  juegos_empatados  juegos_perdidos  goles_favor  \
0                2                 1                0            7   
1                2                 1                0            4   
2                1                 0                2            6   
3                0                 0                3            1   
4                3                 0                0           10   
5                2                 0                1            5   
6                1                 0                2            4   
7                0                 0                3            3   
8                3                 0                0            9   
9                1                 1                1            2   
10               1                 0                2            4   
11               0                 1                2            2   
12               2                 1                0            4   
13               2                 0                1            4   
14               1                 0                2            2   
15               0                 1                2            2   
16               2                 1                0            8   
17               2                 0                1            7   
18               1                 1                1            3   
19               0                 0           

Revisamos si existen valores nulos en el dataframe que puedan afectar en el analisis estadistico

In [146]:
df_worldcup.isnull().sum()

team                0
continent           0
group               0
group_pos           0
cantidad_juegos     0
juegos_ganados      0
juegos_empatados    0
juegos_perdidos     0
goles_favor         0
goles_contra        0
goles_diferencia    0
puntos              0
clasificado         0
dtype: int64

Confirmamos que no hay valores nulos, ahora usamos la funcion describe paara tener un primer approach estaditico del dataframe

In [147]:
df_worldcup.describe()

group_pos  cantidad_juegos  juegos_ganados  juegos_empatados  \
count  32.000000             32.0       32.000000         32.000000   
mean    2.500000              3.0        1.218750          0.562500   
std     1.135924              0.0        1.007532          0.564401   
min     1.000000              3.0        0.000000          0.000000   
25%     1.750000              3.0        0.000000          0.000000   
50%     2.500000              3.0        1.000000          1.000000   
75%     3.250000              3.0        2.000000          1.000000   
max     4.000000              3.0        3.000000          2.000000   

       juegos_perdidos  goles_favor  goles_contra     puntos  clasificado  
count         32.00000    32.000000     32.000000  32.000000    32.000000  
mean           1.21875     4.250000      4.250000   4.218750     0.500000  
std            0.97499     2.314052      2.228843   2.881917     0.508001  
min            0.00000     1.000000      1.000000   0.000000     0.000000  
25%            0.00000     2.750000      3.000000   1.750000     0.000000  
50%            1.00000     4.000000      4.000000   4.000000     0.500000  
75%            2.00000     6.000000      6.000000   7.000000     1.000000  
max            3.00000    10.000000      9.000000   9.000000     1.000000

**1.** Considerando el experimento de seleccionar un país participante al azar, se pide determinar las probabilidades de los siguientes sucesos:

*a.* pertenecer a Africa

In [148]:
total_paises = len(df_worldcup.index)
total_paises_africa = df_worldcup['continent'].value_counts()['africa']
total_paises_asia = df_worldcup['continent'].value_counts()['asia']
total_paises_europa = df_worldcup['continent'].value_counts()['europe']
total_paises_norteamerica= df_worldcup['continent'].value_counts()['northamerica']
total_paises_sudamerica = df_worldcup['continent'].value_counts()['southamerica']

print("total paises: " + str(total_paises))
print("total paises de africa: " + str(total_paises_africa))
print("total paises de asia: " + str(total_paises_asia))
print("total paises de europa: " + str(total_paises_europa))
print("total paises de norteamerica: " + str(total_paises_norteamerica))
print("total paises de sudamerica: " + str(total_paises_sudamerica))

total paises: 32
total paises de africa: 5
total paises de asia: 4
total paises de europa: 13
total paises de norteamerica: 4
total paises de sudamerica: 6


In [149]:
prob_pais_africa = total_paises_africa / total_paises
prob_pais_africa

0.15625

La probabilidad de seleccionar un pais al azar y que este sea de Africa es 0.15625

*b.* pertenecer a Asia

In [150]:
prob_pais_asia = total_paises_asia / total_paises
prob_pais_asia

0.125

La probabilidad de seleccionar un pais al azar y que este sea de Asia es 0.125

*c.* pertenecer a Europa

In [151]:
prob_pais_europa = total_paises_europa / total_paises
prob_pais_europa

0.40625

La probabilidad de seleccionar un pais al azar y que este sea de Europa es 0.40625

*d.* pertenecer a Norteamerica

In [152]:
prob_pais_norteamerica = total_paises_norteamerica / total_paises
prob_pais_norteamerica

0.125

La probabilidad de seleccionar un pais al azar y que este sea de Norteamerica es 0.125

*e.* pertenecer a Sudamerica

In [153]:
prob_pais_sudamerica = total_paises_sudamerica / total_paises
prob_pais_sudamerica

0.1875

La probabilidad de seleccionar un pais al azar y que este sea de Sudamerica es 0.1875

*f.* clasificar a segunda ronda

In [154]:
total_clasificados = df_worldcup['clasificado'].value_counts()[1]
prob_clasificado = total_clasificados / total_paises
prob_clasificado

0.5

La probabilidad de seleccionar un pais al azar y que este sea clasifique a segunda ronda es de 0.5

*g.* ganar al menos un partido

In [155]:
al_menos_un_partido = (df_worldcup['juegos_ganados'] >= 1).sum()
prob_al_menos_un_partido = al_menos_un_partido / total_paises
prob_al_menos_un_partido

0.71875

La probabilidad de seleccionar un pais al azar y que este gane al menos un partido es de 0.71875

*h.* pertenecer a africa y clasificar a segunda ronda

In [156]:
condicion_africa = df_worldcup['continent'] == 'africa'
condicion_europa = df_worldcup['continent'] == 'europe'
condicion_asia = df_worldcup['continent'] == 'asia'
condicion_sudamerica = df_worldcup['continent'] == 'southamerica'
condicion_norteamerica = df_worldcup['continent'] == 'northamerica'

condicion_clasificado = df_worldcup['clasificado'] == 1

conteo_africa_y_clasificado = (condicion_africa & condicion_clasificado).sum()
prob_africa_y_clasificado = conteo_africa_y_clasificado / total_paises
prob_africa_y_clasificado

0.0625

La probabilidad de seleccionar un pais al azar y que este sea de africa y clasifique es de 0.0625

*i.* pertenecer a europa y clasificar a segunda ronda

In [157]:
count_europa_y_clasificado = (condicion_europa & condicion_clasificado).sum()
prob_europa_y_clasificado = count_europa_y_clasificado / total_paises
prob_europa_y_clasificado

0.1875

La probabilidad de seleccionar un pais al azar y que este sea de europa y clasifique es de 0.1875

*j.* pertenecer a europa y clasificar a segunda ronda

In [158]:
count_asia_y_clasificado = (condicion_asia & condicion_clasificado).sum()
prob_asia_y_clasificado = count_asia_y_clasificado / total_paises
prob_asia_y_clasificado

0.0

La probabilidad de seleccionar un pais al azar y que este sea de asia y clasifique es de 0

*k.* pertenecer a sudamerica y clasificar a segunda ronda

In [159]:
count_sudamerica_y_clasificado = (condicion_sudamerica & condicion_clasificado).sum()
prob_sudamerica_y_clasificado = count_sudamerica_y_clasificado / total_paises
prob_sudamerica_y_clasificado

0.15625

La probabilidad de seleccionar un pais al azar y que este sea de sudamerica y clasifique es de 0.15625

*l.* pertenecer a norteamerica y clasificar a segunda ronda

In [160]:
count_norteamerica_y_clasificado = (condicion_norteamerica & condicion_clasificado).sum()
prob_norteamerica_y_clasificado = count_norteamerica_y_clasificado / total_paises
prob_norteamerica_y_clasificado

0.09375

La probabilidad de seleccionar un pais al azar y que este sea de norteamerica y clasifique es de 0.09375

*m.* pertenecer a africa o clasificar a segunda ronda

In [161]:
conteo_africa_o_clasificado = (condicion_africa | condicion_clasificado).sum()
prob_africa_o_clasificado = conteo_africa_o_clasificado / total_paises
prob_africa_o_clasificado

0.59375

In [162]:
#Tambien se puede calcular como:
(prob_pais_africa + prob_clasificado) - prob_africa_y_clasificado

0.59375

La probabilidad de seleccionar un pais al azar y que este sea de africa o clasifique es 0.59375

*n.* pertenecer a europa o clasificar a segunda ronda

In [163]:
conteo_europa_o_clasificado = (condicion_europa | condicion_clasificado).sum()
prob_europa_o_clasificado = conteo_europa_o_clasificado / total_paises
prob_europa_o_clasificado

0.71875

In [164]:
#tambien se puede calcular como:
(prob_pais_europa + prob_clasificado) - prob_europa_y_clasificado

0.71875

La probabilidad de seleccionar un pais al azar y que este sea de europa o clasifique es 0.71875

*o.* pertenecer a asia o clasificar a segunda ronda

In [165]:
conteo_asia_o_clasificado = (condicion_asia | condicion_clasificado).sum()
prob_asia_o_clasificado = conteo_asia_o_clasificado / total_paises
prob_asia_o_clasificado

0.625

In [166]:
#tambien se puede calcular como:
(prob_pais_asia + prob_clasificado) - prob_asia_y_clasificado

0.625

La probabilidad de seleccionar un pais al azar y que este sea de asia o clasifique es 0.625

*p.* pertenecer a sudamerica o clasificar a segunda ronda

In [167]:
conteo_sudamerica_o_clasificado = (condicion_sudamerica | condicion_clasificado).sum()
prob_sudamerica_o_clasificado = conteo_sudamerica_o_clasificado / total_paises
prob_sudamerica_o_clasificado

0.53125

In [168]:
#tambien se puede calcular como:
(prob_pais_sudamerica + prob_clasificado) - prob_sudamerica_y_clasificado

0.53125

La probabilidad de seleccionar un pais al azar y que este sea de sudamerica o clasifique es 0.53125

*q.* pertenecer a norteamerica o clasificar a segunda ronda

In [169]:
conteo_norteamerica_o_clasificado = (condicion_norteamerica | condicion_clasificado).sum()
prob_norteamerica_o_clasificado = conteo_norteamerica_o_clasificado / total_paises
prob_norteamerica_o_clasificado

0.53125

In [170]:
#tambien se puede calcular como:
(prob_pais_norteamerica + prob_clasificado) - prob_norteamerica_y_clasificado

0.53125

La probabilidad de seleccionar un pais al azar y que este sea de norteamerica o clasifique es 0.53125

*r.* pertenecer a africa y haber ganado al menos un partido

In [171]:
condicion_partido = df_worldcup['juegos_ganados'] >= 1

conteo_africa_y_partido = (condicion_africa & condicion_partido).sum()
prob_africa_y_partido = conteo_africa_y_partido / total_paises
prob_africa_y_partido

0.09375

La probabilidad de seleccionar un pais al azar y que este sea de africa y haya ganado al menos un partido es 0.09375

*s.* pertenecer a europa y haber ganado al menos un partido

In [172]:
conteo_europa_y_partido = (condicion_europa & condicion_partido).sum()
prob_europa_y_partido = conteo_europa_y_partido / total_paises
prob_europa_y_partido

0.34375

La probabilidad de seleccionar un pais al azar y que este sea de europa y haya ganado al menos un partido es 0.34375

*t.* pertenecer a asia y haber ganado al menos un partido

In [173]:
conteo_asia_y_partido = (condicion_asia & condicion_partido).sum()
prob_asia_y_partido = conteo_asia_y_partido / total_paises
prob_asia_y_partido

0.0

La probabilidad de seleccionar un pais al azar y que este sea de asia y haya ganado al menos un partido es 0

*u.* pertenecer a sudamerica y haber ganado al menos un partido

In [174]:
conteo_sudamerica_y_partido = (condicion_sudamerica & condicion_partido).sum()
prob_sudamerica_y_partido = conteo_sudamerica_y_partido / total_paises
prob_sudamerica_y_partido

0.1875

La probabilidad de seleccionar un pais al azar y que este sea de sudamerica y haya ganado al menos un partido es 0.1875

*v.* pertenecer a norteamerica y haber ganado al menos un partido

In [175]:
conteo_norteamerica_y_partido = (condicion_norteamerica & condicion_partido).sum()
prob_norteamerica_y_partido = conteo_norteamerica_y_partido / total_paises
prob_norteamerica_y_partido

0.09375

La probabilidad de seleccionar un pais al azar y que este sea de norteamerica y haya ganado al menos un partido es 0.09375

*w.* haber ganado al menos un partido y clasificar a segunda ronda

In [176]:
conteo_partido_y_clasificado = (condicion_partido & condicion_clasificado).sum()
prob_partido_y_clasificado = conteo_partido_y_clasificado / total_paises
prob_partido_y_clasificado

0.5

La probabilidad de seleccionar un pais al azar y que este haya ganado al menos un partido y haya clasidicado a segunda ronda es 0.5